### PR TITLE
Feature: Added Modals, Added ability to add full detail of a model when has relationships

### PR DESCRIPTION
**Pull request by Xoshbin**
Voyager by default when a model have a relationship it adds only the field that's been added when making the bread. But this PR adds the ability to show a button beside those fields that have relationship and when clicked showing the complete form of that related bread. It's been tested all working well when adding filling the form it saves the data without leaving the page. what is missing is a javascript or ajax function to update the current row that have the button beside.

### CODE REVIEW

**Comment by acicovic** [comment]
Hi @Xoshbin, can you provide a quick walkthrough on how to make this work? I tried using it but saw no buttons or modals.

**Comment by Xoshbin** [comment]
Hi @acicovic , I just tested it on a new and clean project the only thing it need to change was on modals\create-blade.php where the save button was giving error. and I found out that it only works for generated breads so it's not working on "posts" bread which it has a custom view.

**Comment by acicovic** [comment]
Hi,

For me the button located at line 76 of resources/views/bread/edit-add.blade.php never appears on forms. So for some reason I am not hitting it. Do I have to declare the relationships in the models or should the relationships declared in Voyager's interface work ?

Thanks!

**Comment by acicovic** [comment]
The button works now.

**Comment by Xoshbin** [comment]
@acicovic It looks like it's working for you now. Yes you have to make the relationships from the bread generator then it's going to display. Sorry for late reply.

**Comment by tiagopimentelgomes** [comment]
"what is missing is a javascript or ajax function to update the current row that have the button beside".

First, thank you Xoshbin for the improvement.

To complete the dropdown's update:

**1) \vendor\tcg\voyager\resources\views\bread\edit-add.blade.php**
Add the javascript just after AddModel(column) function:

> 
        function saveModel(column, urlpath) {
            if ($("#form-create-"+column).length > 0) {
                $.ajax({
                        data: $('#form-create-'+column).serialize(),
                        url: urlpath,
                        type: "POST",
                        dataType: 'json',
                        success: function (data) {
                            $('#form-create-'+column).trigger("reset");
                            $('#modal-create-'+column).modal('hide');
                            $('#modal-create-'+column).detach();
                        },
                        error: function (data) {
                            console.log('Error:', data);
                            $("#button-"+column).html('ERRO');
                        }
                    });
            }
        }

**2) \vendor\tcg\voyager\resources\views\modals\create.blade.php**
Replace the div "panel-footer":

> 

        <div class="panel-footer">
                 <button type="button" 
                          onclick="saveModel('{{ $dataType->slug }}','{{ route('voyager.'.$dataType->slug.'.store') }}')" 
                          id="button-{{ $dataType->slug }}" 
                          class="btn btn-primary save"
                          tabindex="-1">
                          {{ __('voyager::generic.save') }}
                 </button>
        </div>

**Comment by Xoshbin** [comment]
Hey @tiagopimentelgomes Thank you so much I will test it out for sure.

**Comment by srigurubyo** [comment]
Any update on this?
